{
  "uid" : "6130e520446be335",
  "name" : "NHS Number: Searching with invalid NHS number shows an error [123456789-Enter 10 digits]",
  "fullName" : "/app/features/find_a_patient.feature:NHS Number: Searching with invalid NHS number shows an error",
  "historyId" : "f44352c612938e33e618f3f1b20aabf5",
  "time" : {
    "start" : 1728478115711,
    "stop" : 1728478125424,
    "duration" : 9713
  },
  "status" : "failed",
  "statusMessage" : "TypeError: get_element_by_type() takes from 2 to 3 positional arguments but 4 were given",
  "statusTrace" : "fixturefunc = <function step_error_message_appears_for_nhs_number at 0x7fcbf06cbf60>\nrequest = <FixtureRequest for <Function test_nhs_number_searching_with_invalid_nhs_number_shows_an_error[123456789-Enter 10 digits]>>\nkwargs = {'errorMessage': 'Enter 10 digits'}\n\n    def call_fixture_func(\n        fixturefunc: _FixtureFunc[FixtureValue], request: FixtureRequest, kwargs\n    ) -> FixtureValue:\n        if is_generator(fixturefunc):\n            fixturefunc = cast(\n                Callable[..., Generator[FixtureValue, None, None]], fixturefunc\n            )\n            generator = fixturefunc(**kwargs)\n            try:\n                fixture_result = next(generator)\n            except StopIteration:\n                raise ValueError(f\"{request.fixturename} did not yield a value\") from None\n            finalizer = functools.partial(_teardown_yield_fixture, fixturefunc, generator)\n            request.addfinalizer(finalizer)\n        else:\n            fixturefunc = cast(Callable[..., FixtureValue], fixturefunc)\n>           fixture_result = fixturefunc(**kwargs)\n\n.tox/py311/lib/python3.11/site-packages/_pytest/fixtures.py:898: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \ntests/test_find_a_patient_steps.py:126: in step_error_message_appears_for_nhs_number\n    assert check_nhs_number_error_message_text_exists(errorMessage) is True\npages/find_a_patient_page.py:121: in check_nhs_number_error_message_text_exists\n    return check_if_element_exists(\"text\", f\"Error: {errorMessage}\")\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nelement = 'text', wait = 'Error: Enter 10 digits'\n\n    def check_if_element_exists(element, wait=False):\n>       element = get_element_by_type(*element)\nE       TypeError: get_element_by_type() takes from 2 to 3 positional arguments but 4 were given\n\ninit_helpers.py:142: TypeError",
  "flaky" : false,
  "newFailed" : false,
  "newBroken" : false,
  "newPassed" : false,
  "retriesCount" : 0,
  "retriesStatusChange" : false,
  "beforeStages" : [ ],
  "testStage" : {
    "status" : "failed",
    "statusMessage" : "TypeError: get_element_by_type() takes from 2 to 3 positional arguments but 4 were given",
    "statusTrace" : "fixturefunc = <function step_error_message_appears_for_nhs_number at 0x7fcbf06cbf60>\nrequest = <FixtureRequest for <Function test_nhs_number_searching_with_invalid_nhs_number_shows_an_error[123456789-Enter 10 digits]>>\nkwargs = {'errorMessage': 'Enter 10 digits'}\n\n    def call_fixture_func(\n        fixturefunc: _FixtureFunc[FixtureValue], request: FixtureRequest, kwargs\n    ) -> FixtureValue:\n        if is_generator(fixturefunc):\n            fixturefunc = cast(\n                Callable[..., Generator[FixtureValue, None, None]], fixturefunc\n            )\n            generator = fixturefunc(**kwargs)\n            try:\n                fixture_result = next(generator)\n            except StopIteration:\n                raise ValueError(f\"{request.fixturename} did not yield a value\") from None\n            finalizer = functools.partial(_teardown_yield_fixture, fixturefunc, generator)\n            request.addfinalizer(finalizer)\n        else:\n            fixturefunc = cast(Callable[..., FixtureValue], fixturefunc)\n>           fixture_result = fixturefunc(**kwargs)\n\n.tox/py311/lib/python3.11/site-packages/_pytest/fixtures.py:898: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \ntests/test_find_a_patient_steps.py:126: in step_error_message_appears_for_nhs_number\n    assert check_nhs_number_error_message_text_exists(errorMessage) is True\npages/find_a_patient_page.py:121: in check_nhs_number_error_message_text_exists\n    return check_if_element_exists(\"text\", f\"Error: {errorMessage}\")\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nelement = 'text', wait = 'Error: Enter 10 digits'\n\n    def check_if_element_exists(element, wait=False):\n>       element = get_element_by_type(*element)\nE       TypeError: get_element_by_type() takes from 2 to 3 positional arguments but 4 were given\n\ninit_helpers.py:142: TypeError",
    "steps" : [ {
      "name" : "Given I am on the find a patient by nhs number page",
      "time" : {
        "start" : 1728478115712,
        "stop" : 1728478123382,
        "duration" : 7670
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "And I enter 123456789 as the nhs number",
      "time" : {
        "start" : 1728478123383,
        "stop" : 1728478125201,
        "duration" : 1818
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "When I click the search button",
      "time" : {
        "start" : 1728478125201,
        "stop" : 1728478125287,
        "duration" : 86
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "Then I can see an nhs number error message Enter 10 digits",
      "time" : {
        "start" : 1728478125287,
        "stop" : 1728478125424,
        "duration" : 137
      },
      "status" : "failed",
      "statusMessage" : "get_element_by_type() takes from 2 to 3 positional arguments but 4 were given",
      "statusTrace" : "TypeError: get_element_by_type() takes from 2 to 3 positional arguments but 4 were given\n",
      "steps" : [ ],
      "attachments" : [ {
        "uid" : "10ae9fe847facade",
        "name" : "dev_chrome_129.0.6668.100_error_message_appears_for_nhs_number.png",
        "source" : "10ae9fe847facade.png",
        "type" : "image/png",
        "size" : 69214
      } ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 1,
      "shouldDisplayMessage" : true,
      "hasContent" : true
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 4,
    "attachmentsCount" : 1,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ ],
  "labels" : [ {
    "name" : "host",
    "value" : "8121528f60b7"
  }, {
    "name" : "thread",
    "value" : "1046-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest-bdd"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "feature",
    "value" : "Find a patient"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "errorMessage",
    "value" : "Enter 10 digits"
  }, {
    "name" : "nhsNumber",
    "value" : "123456789"
  } ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Product defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "tags" : [ ]
  },
  "source" : "6130e520446be335.json",
  "parameterValues" : [ "Enter 10 digits", "123456789" ]
}